geode-core/src/distributedTest/java/org/apache/geode/cache30/CacheStatisticsPartitionedRegionDUnitTest.java

```diff
@@ -0,0 +1,577 @@
+      }
+    });
+
+    vm0.invoke(new CacheSerializableRunnable("Invalidate") {
+      @Override
+      public void run2() throws CacheException {
+        Region region = getRootRegion(name);
+        CacheStatistics stats = region.getStatistics();
+        lastAccessed = stats.getLastAccessedTime();
+        lastModified = stats.getLastModifiedTime();
+        region.invalidate(key);
+        region.invalidate(key1);
+
+        GeodeAwaitility.await().atMost(MSECS, TimeUnit.MILLISECONDS)
+            .until(() -> stats.getLastAccessedTime() == lastAccessed);
+        assertEquals(lastModified, stats.getLastModifiedTime());
+        assertEquals(4, stats.getHitCount());
+        assertEquals(4, stats.getMissCount());
+      }
+    });
+
+    waitForOp();
+
+    vm1.invoke(new CacheSerializableRunnable("Invalidate") {
+      @Override
+      public void run2() throws CacheException {
+        Region region = getRootRegion(name);
+        CacheStatistics stats = region.getStatistics();
+        lastAccessed = stats.getLastAccessedTime();
+        lastModified = stats.getLastModifiedTime();
+        region.invalidate(key2);
+        region.invalidate(key3);
+
+        GeodeAwaitility.await().atMost(MSECS, TimeUnit.MILLISECONDS)
+            .until(() -> stats.getLastAccessedTime() == lastAccessed);
+        assertEquals(lastModified, stats.getLastModifiedTime());
+        assertEquals(4, stats.getHitCount());
+        assertEquals(4, stats.getMissCount());
+      }
+    });
+
+    vm0.invoke(new CacheSerializableRunnable("Destroy Entry") {
+      @Override
+      public void run2() throws CacheException {
+        Region region = getRootRegion(name);
+        CacheStatistics stats = region.getStatistics();
+        lastModified = stats.getLastModifiedTime();
+        region.destroy(key);
+        region.destroy(key1);
+
+        GeodeAwaitility.await().atMost(MSECS, TimeUnit.MILLISECONDS)
+            .until(() -> stats.getLastModifiedTime() == lastModified);
+        assertEquals(4, stats.getHitCount());
+        assertEquals(4, stats.getMissCount());
+      }
+    });
+
+    waitForOp();
+
+    vm1.invoke(new CacheSerializableRunnable("Destroy Entry") {
+      @Override
+      public void run2() throws CacheException {
+        Region region = getRootRegion(name);
+        CacheStatistics stats = region.getStatistics();
+        lastModified = stats.getLastModifiedTime();
+        region.destroy(key2);
+        region.destroy(key3);
+
+        GeodeAwaitility.await().atMost(MSECS, TimeUnit.MILLISECONDS)
+            .until(() -> stats.getLastModifiedTime() == lastModified);
+        assertEquals(4, stats.getHitCount());
+        assertEquals(4, stats.getMissCount());
+      }
+    });
+
+  }
+
+  /**
+   * Tests that an attempt to get statistics when they are disabled results in a
+   * {@link StatisticsDisabledException}.
+   */
+  @Test
+  public void testDisabledStatistics() throws CacheException {
```

Comment:
As the method is so short, I would rather leave just one method to check that the Exception is raised when trying to get the statistics from the region and from the entry.
By the way, there was an error in the test case. Instead of getting the stats for the region and then for the entry, I got twice the stats for the entry.
I agree on changing the way exceptions are asserted by using assertThatThrows.